tests/unit/tests_high_five_base.cpp

```diff
@@ -28,6 +28,15 @@
 #include <highfive/highfive.hpp>
 #include "tests_high_five.hpp"
 
+#ifdef H5_USE_BOOST
+#include <highfive/boost.hpp>
```

Comment:
Just confirming the intent: these headers are not included by default, so it is users' responsibility to meet the dependencies if they want to use/include them? That is cool

Reply:
Exactly, same mechanism that pybind11 and boost use to give you default specializations of traits for STL containers.